### ISSUE TITLE
Module info

### ISSUE THREAD

**Issue by bestouff** [quadrupleslap/badlog#3]
(Yes, me again)
`env_logger` prints module names by default, which can be quite handy especially when you use lots of modules. Would it be possible to do the same ?
Here's an `env_logger` example:
```
INFO:ntpsek: Filtering packets from NTP server 172.16.5.202:123
INFO:ntpsek: Connecting to MQTT server 172.16.5.10:1883
INFO:mqttc::client:  Connecting to 172.16.5.10:1883
INFO:mqttc::client:     Connection accepted
INFO:ntpsek: Fetching packets from eth1
```
Here's the same with `bad_logger`:
```
 [INFO] Filtering packets from NTP server 172.16.5.202:123
 [INFO] Connecting to MQTT server 172.16.5.10:1883
 [INFO]  Connecting to 172.16.5.10:1883
 [INFO]     Connection accepted
 [INFO] Fetching packets from eth1
```
Maybe the module part should be painted in yet another color (blue) not to mix it with the rest.
(Yes I know, the alignment would be lost)

**Comment by quadrupleslap** [comment]
Any idea on how to do this while retaining the current simplicity?

**Comment by bestouff** [comment]
Just always do it ? I don't see much harm in always printing the module name, but maybe I'm biased.
(That said probably file name and line number are too much)

**Comment by bestouff** [comment]
Or you could try to enable that through [`Cargo.toml`'s "features"](http://doc.crates.io/manifest.html#the-features-section). That's the least invasive I can think of.

**Comment by quadrupleslap** [comment]
Okay, this sounds a *bit* like an abuse of cargo features, but I guess it's the best option, unless we want to enable or disable this at runtime, too.

So, what format? I was thinking:

```
[hh:mm:ss]? [module]? [level] message
[19:20:11] [mqttc::client] [INFO] Connecting to 172.16.5.10:1883
```

Where only the `[level]` is colored.

**Comment by bestouff** [comment]
I would put the variable-length field (module name) last to keep everything aligned as much as possible. So [date] [time] [level] [module] etc.

**Comment by quadrupleslap** [comment]
1. Time and module should be greyed out (`Fixed(8)`).
2. Should module be padded out to the size of the longest previously logged module? I'm leaning towards no.

**Comment by quadrupleslap** [comment]
This looks pretty good:

![](http://i.imgur.com/MRgXfrVl.png)

**Comment by bestouff** [comment]
I would have preferred not touching the normal text (in case there is some colored text in the log itself). Otherwise it's perfect !

**Comment by quadrupleslap** [comment]
Oh, that's just my terminal's default coloring lol.

**Comment by quadrupleslap** [comment]
Closed by 8f78bbf1dd389cf1a6867061b60397f695e70185.

**Comment by bestouff** [comment]
Looks perfect now, and still simple. Thanks !
